### PR TITLE
New layout: FlexBoxLayout

### LINKED ISSUE #66: Add support for FlexBox layout
Make it possible to use layouts in `.60` files that are based on the CSS flex box layout.

Stretch offers itself as an implementation in Rust for that purpose.

Acceptance criteria: Flex layouts can be used in `.60` files and their API is documented.

### PR DESCRIPTION
**Pull request by dfaure-kdab**
This PR adds FlexBoxLayout to Slint—a flexible layout system that arranges children in rows or columns with automatic wrapping, inspired by CSS Flexbox.

# Features
4 directions: row, column, row-reverse, column-reverse (runtime-switchable)
Wrapping: Items wrap to new rows/columns when space exceeded
Spacing & padding: Configurable gaps and margins (per-side or uniform)
Height-for-width: Layout adapts responsively based on available space
Repeater support: Works with for loops and dynamic content
Min/max constraints: Respects width/height constraints on items

# Implementation
Integrates [Taffy](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) (v0.6) for flexbox calculations
Compile-time optimization when direction is constant

# Testing
- 17 automated tests covering wrapping, spacing, constraints, repeaters, runtime direction changes, nested layouts, and edge cases.
- PrinterDemo ported to use FlexBoxLayout
- An interactive example in examples/layouts/

# Not implemented yet
- align-items, justify-content, flex-wrap, align-content, and more properties from the CSS spec

Fixes #66

### CODE REVIEW

File: internal/compiler/llr/lower_expression.rs [hunk 1]
```diff
@@ -888,6 +890,180 @@ fn solve_box_layout(
     }
 }
```
Comment (ogoffart) [inline internal/compiler/llr/lower_expression.rs:892]: (Doesn't have to be in this PR, but the layout code is starting to be quite big in this module.  maybe all the layout things could be extracted to llr/lower_layout.rs)

File: docs/astro/src/content/docs/reference/layouts/flexboxlayout.mdx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+## Overview
+
+In row direction, items are placed from left to right. When the available width is exceeded, items automatically wrap to the next row. In column direction, items are placed from top to bottom and wrap to the next column when the available height is exceeded.
+
+## Spacing Properties
+
+### spacing
+<SlintProperty propName="spacing" typeName="length">
+The distance between the elements in the layout. CSS Flexbox usually calls this "gap", but "spacing" is used in Slint for consistency with other layout types.
```
Comment (ogoffart) [inline docs/astro/src/content/docs/reference/layouts/flexboxlayout.mdx:62]: Idea: Have a gap property but have an error when it is used to tell user to use the spacing property.  (Alternatively, could also be a deprecated alias, but it feels strange to add a deprecated property)

File: docs/astro/src/content/docs/reference/layouts/flexboxlayout.mdx
Comment (ogoffart) [inline docs/astro/src/content/docs/reference/layouts/flexboxlayout.mdx:60]: In builtins.slint , i see there is also horizontal and vertical spacing.

File: internal/compiler/llr/lower_expression.rs (hunk 1 above)
Comment (dfaure-kdab) [reply internal/compiler/llr/lower_expression.rs:892]: Good idea, but there's already passes/lower_layout.rs, so maybe it would be better with a different filename? `lower_layout_expressions.rs?`

File: docs/astro/src/content/docs/reference/layouts/flexboxlayout.mdx
Comment (dfaure-kdab) [reply docs/astro/src/content/docs/reference/layouts/flexboxlayout.mdx:60]: indeed (and `tests/cases/layout/flexbox_spacing_and_padding.slint` checks that it works)

**Comment by dfaure-kdab** [comment]
> I skimmed through this big change and it looks good.

Yeah, sorry about the big size. I didn't find a good way to split it up.

> The only blocker I spotted is the missing documentation for the horizontal and vertical spacing.

Well spotted. Added.

File: docs/astro/src/content/docs/reference/layouts/flexboxlayout.mdx (hunk 2 above)
Comment (dfaure-kdab) [reply docs/astro/src/content/docs/reference/layouts/flexboxlayout.mdx:62]: I thought users would search the documentation for "gap" (as well as row-gap and column-gap which I just added), when noticing it wouldn't work. But indeed I can make it a compile error (in a followup PR).

File: internal/compiler/llr/lower_expression.rs (hunk 1 above)
Comment (dfaure-kdab) [reply internal/compiler/llr/lower_expression.rs:892]: => https://github.com/slint-ui/slint/pull/10775

File: docs/astro/src/content/docs/reference/layouts/flexboxlayout.mdx (hunk 2 above)
Comment (dfaure-kdab) [reply docs/astro/src/content/docs/reference/layouts/flexboxlayout.mdx:62]: => https://github.com/slint-ui/slint/pull/10776
